src/spectralanalysis.jl

```diff
@@ -158,4 +158,54 @@ Nmax (optional)
 """
 groundstate(H::Union{DenseOperator, SparseOperator}) = eigenstates_hermitian(H; Nmax=1)[1]
 
+
+"""
+Simultaneously diagonalize commuting Hermitian operators.
+
+This is done by diagonalizing the sum of the operators.
+The eigenvalues are computed by :math:`a = \\langle \\psi |A|\\psi\\rangle` and
+it is checked whether the eigenvectors fulfill the equation
+:math:`A|\\psi\\rangle = a|\\psi\\rangle`.
+
+Arguments
+---------
+
+Ops
+  Vector of operators (sparse or dense).
+
+Keyword arguments
+-----------------
+
+atol (optional)
+  kwarg of Base.isapprox specifying the tolerance of the approximate check
+  Default is 1e-14.
+
+rtol (optional)
+  kwarg of Base.isapprox specifying the tolerance of the approximate check
+  Default is 1e-14.
+"""
+
+function simdiag{T <: DenseOperator}(Ops::Vector{T}; atol::Real=1e-14, rtol::Real=1e-14)
+
+  # Check input
+  for A=Ops
+    A == dagger(A) ? nothing : error("Non-hermitian operator given!")
+  end
+
+  d, v = eig(sum(Ops).data)
+
+  evals = [Vector{Complex128}(length(d)) for i=1:length(Ops)]
+  for i=1:length(Ops), j=1:length(d)
+    evals[i][j] = (v[:, j]'*Ops[i].data*v[:, j])[1]
+    check_eval = isapprox(Ops[i].data*v[:, j] - evals[i][j]*v[:, j], zeros(length(d)); atol=atol, rtol=rtol)
+    check_eval ? nothing : error("Simultaneous diagonalization failed!")
```

Comment:
Why compare it to a zeros array. Can't we do isapprox(a, b; ...)
And I think it's nicer to do
if !isapprox(...)
   error(...)
end